### ISSUE TITLE
[React-Navigation 7.x bug] useHeaderConfigProps.js -- "Cannot read property 'regular' of undefined"

### ISSUE THREAD

**Issue by Nantris** [react-navigation/react-navigation#12222]
### Current behavior

"Cannot read property 'regular' of undefined", seemingly due to this line: https://github.com/react-navigation/react-navigation/blob/main/packages/native-stack/src/views/useHeaderConfigProps.tsx#L59

Not sure why `fonts` would be undefined.

### Expected behavior

No error

### Reproduction

N/A

### Platform

- [x] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [x] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [x] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native-stack         | 7.0.0-rc.30

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I've identified the issue in the `useHeaderConfigProps` function where `fonts` could be undefined. To fix this, I've added a fallback for when `fonts` is undefined. This ensures that we always have a valid object to work with, preventing the 'Cannot read property 'regular' of undefined' error.

Here's what I've changed in the `useHeaderConfigProps` function:

I also created pull request: https://github.com/react-navigation/react-navigation/pull/12223


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

**Comment by satya164** [comment]
The only way `fonts` could be `undefined` is if you pass an invalid theme. Please provide a repro for the issue to identify if there's a bug in the code.

**Comment by Nantris** [comment]
I see I can't read and this was indeed mentioned: https://reactnavigation.org/docs/7.x/upgrading-from-6.x/#the-options-callback-gets-theme

**Comment by Nantris** [comment]
@satya164 I think the PR mentioned above is worthwhile to avoid this issue though.

**Comment by satya164** [comment]
@Nantris I don't think it has to do anything with `options` callback. You're not passing the `fonts` property https://reactnavigation.org/docs/7.x/upgrading-from-6.x/#the-theme-prop-now-accepts-a-fonts-property

**Comment by Nantris** [comment]
@satya164 I think it would be good to do something like `{ ...DefaultTheme, ...userTheme }` behind the scenes to make it impossible to break this while updating so it would "just work" 

But this is a nice-to-have at best and adds some code clutter.

Somewhere just after here it would be inserted, if desired: https://github.com/react-navigation/react-navigation/blob/581681aea465b871728b9d7e4b1a6eebbcf05ed0/packages/native/src/NavigationContainer.tsx#L67
